chore (refs T33323): add storybook for DpSelect component. (#350)
demos-europe/demosplan-ui: src/components/DpSelect/DpSelect.stories.mdx

import { Canvas, Meta, Story } from '@storybook/addon-docs'
import DpSelect from './DpSelect'

<Meta
  title="Components/Select"
  component={DpSelect} />

# DpSelect

The DpSelect component is used to render a select input field with customizable options. It provides functionality for selecting an option from a dropdown list and emits a select event with the selected value.

> 
> ## Default usage
> 
> ```html
> <dp-select
>   v-model="selectModel"
>   :label="{
>           text: Translator.trans('Text')
>         }"
>   name="select_menu_name"
>   :options="selectOptions"
>   required
>   @select="updateLocalState" />
> ```

Reviewer: We do not need to document the html form, if we use the Storybook component implementation.

```suggestion

## Examples
```

export const Template = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { DpSelect },
    template: `<dp-select v-bind="$props" />`
});

### Classes
<Canvas>
    <Story
        name="Classes"
        args={{
            id: 'selectId',
            label: {
              text: 'select label'
            },
            classes: 'layout__item u-6-of-12 sbdocs-preview css-tmvzag'
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

> 

Reviewer: There is only one usage of `classes` within demosplan-core. As this is an ugly opportunity to reach into the internals of the component, i'd suggest to not document it here, rather add the following in line 61 of DpSelect:

```
    /**
     * @deprecated
     */
```

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/350/commits/1bbcb01fa0d35f8ef47d5093c513a3ca3e69604f

### Disabled

<Canvas>
    <Story
        name="Disabled"
        args={{
            id: 'selectId',
            label: {

>               text: 'select label'

Reviewer: For all examples: i think this label is a better fit, especially for the options below.

```suggestion
              text: 'Chose a number'
```

            },
            disabled: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Required

If a select is `required`, an asterisk will be added to its label.

<Canvas>
    <Story
        name="Required"
        args={{
            id: 'selectId',
            label: {
              text: 'select label'
            },
            required: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Options

> Need label and value

Reviewer: ```suggestion
Options are passed as an array of objects, which must have `label` and `value` properties.
```

<Canvas>
    <Story
        name="Options"
        args={{
            id: 'selectId',
            label: {
              text: 'select label'
            },
            options: [
                { label: 'Option 1', value: '1' },
                { label: 'Option 2', value: '2' }
            ]
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

> ### Selected
> 

Reviewer: ```suggestion
### Selected

Pass the value of the option that should be selected.

```

<Canvas>
    <Story
        name="Selected"
        args={{
            id: 'selectId',
            label: {
              text: 'select label'
            },
            options: [
                { label: 'Option 1', value: '1' },
                { label: 'Option 2', value: '2' }
            ],

>             selected: 'selected'

Reviewer: ```suggestion
            selected: '2'
```

import { Meta } from '@storybook/addon-docs'
import DpSelect from './DpSelect'

<Meta
  title="Components/Select"
  component={DpSelect} />

# DpSelect

> The DpSelect component provides `<select>` element to create a drop-down list.
> 
> With `label` attribute can select menu title be passed, with `options` attribute can call a computed.
> 
> And with `@select` event can call a method.

Reviewer: ```suggestion
The DpSelect component is used to render a select input field with customizable options. It provides functionality for selecting an option from a dropdown list and emits a select event with the selected value.
```

## Default usage

```html
<dp-select
  v-model="selectModel"

>   class="u-mb-0_5"

Reviewer: the class does not add value in the example.

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/350/commits/fcd7516cb6e2df5ee35e745788547fcfc2a1d57c

>   label="Select Title"

Reviewer: the `label` prop is not a plain string but an object. See DpLabel, and DpSelect implementation.

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/350/commits/fcd7516cb6e2df5ee35e745788547fcfc2a1d57c

  name="select_menu_name"

>   :options="Computed Name"

Reviewer: Here the example should show some real options in an object. see implementation.

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/350/commits/fcd7516cb6e2df5ee35e745788547fcfc2a1d57c

  required

>   @select="Method Name" />

Reviewer: ```suggestion
  @select="updateLocalState" />
```

import { Canvas, Meta, Story } from '@storybook/addon-docs'
import DpSelect from './DpSelect'

<Meta
  title="Components/Select"
  component={DpSelect} />

# DpSelect

The DpSelect component is used to render a select input field with customizable options. It provides functionality for selecting an option from a dropdown list and emits a select event with the selected value.

## Examples

export const Template = (args, { argTypes }) => ({
    props: Object.keys(argTypes),
    components: { DpSelect },
    template: `<dp-select v-bind="$props" />`
});

### Disabled

<Canvas>
    <Story
        name="Disabled"
        args={{
            id: 'selectId',
            label: {
              text: 'Chose a number'
            },
            disabled: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Required

If a select is `required`, an asterisk will be added to its label.

<Canvas>
    <Story
        name="Required"
        args={{
            id: 'selectId',
            label: {

>               text: 'select label'

Reviewer: ```suggestion
              text: 'Choose a number'
```

            },
            required: true
        }}>
        {Template.bind({})}
    </Story>
</Canvas>

### Options

Options are passed as an array of objects, which must have `label` and `value` properties.

<Canvas>
    <Story
        name="Options"
        args={{
            id: 'selectId',
            label: {

>               text: 'select label'

Reviewer: ```suggestion
              text: 'Choose a number'
```
